src/axom/primal/geometry/BezierCurve.hpp

```diff
@@ -232,26 +338,57 @@ class BezierCurve
     const int ord = getOrder();
     axom::Array<T> dCarray(ord + 1);
 
-    // Run de Casteljau algorithm on each dimension
-    for(int i = 0; i < NDIMS; ++i)
+    // If curve is rational
+    if(m_weights[0] > 0)
     {
-      for(int p = 0; p <= ord; ++p)
+      axom::Array<T> dWarray(ord + 1);
+
+      // Run algorithm from Farin '83 on each dimension
+      for(int i = 0; i < NDIMS; ++i)
       {
-        dCarray[p] = m_controlPoints[p][i];
+        for(int p = 0; p <= ord; ++p)
+        {
+          dCarray[p] = m_controlPoints[p][i] * m_weights[p];
+          dWarray[p] = m_weights[p];
+        }
+
+        for(int p = 1; p <= ord; ++p)
+        {
+          const int end = ord - p;
+          for(int k = 0; k <= end; ++k)
+          {
+            dCarray[k] = lerp(dCarray[k], dCarray[k + 1], t);
+            dWarray[k] = lerp(dWarray[k], dWarray[k + 1], t);
+          }
+        }
+        ptval[i] = dCarray[0] / dWarray[0];
```

Comment:
Do we have do guard against division by zero?

Reply:
The recursive algorithms from [Farin 83] assume positive weights at each node. Is that too strong of an assumption for rationals generated from NURBS curves? 

In this case, even if we allow for nonnegative weights in general, the value of `dWarray[0]` can only mathematically be zero if the weight at one or both of the endpoints is zero, which shouldn't be allowed in general.

Reply:
For the time being at least, I've just made the requirements that all weights be positive, as it makes the computation in `.split()` more straightforward in the end, but this could change in the future.

Reply:
Thanks. I agree that, assuming that weights must be strictly positive, we should be ok w.r.t. division by zero.